lib/import/helpers/brca/providers/rth/constants.rb

```diff
@@ -64,14 +64,10 @@ module Constants
 
             CDNA_REGEX = /c\.\[?(?<cdna>[0-9]+.+[a-z]+)\]?/i
 
-            EXON_REGEX = /(?<variant>del|inv|dup).+ion\s(?:of\s)?
-                         exons?\s?(?<location>[0-9]+(?:-[0-9]+)?)|
-                         ex(?:on)?\s?(?<location>[0-9]+(?:-[0-9]+)?)\s?(?<variant>del|inv|dup)|
-                         exons?\s?(?<location>[0-9]+(?:-[0-9]+)?)\s?(?<variant>del|inv|dup)|
-                         ex\s?(?<location>[0-9]+\+[0-9]+)\s?(?<variant>del|inv|dup)|
-                         (?<variant>del|inv|dup).+ion\sBRCA1\sexons?\s(?<location>[0-9]+-(?:[0-9]*))|
-                         exon\s?(?<location>[0-9]+)\s?-exon\s?(?<moreexon>[0-9]+)
-                         \s?(?<variant>del|inv|dup)/ix
+            EXON_REGEX = /(?<mutationtype>del|inv|dup).+ion\s(?:of\s)?exons?\s?(?<exons>[0-9]+(?:-[0-9]+)?)|
+                          ex(?:on|ons)?\s?(?<exons>[0-9]+(?:(?:-|\+)[0-9]+)?)\s?(?<mutationtype>del|inv|dup)|
+                          (?<mutationtype>del|inv|dup).+ion\sBRCA1\sexons?\s(?<exons>[0-9]+-(?:[0-9]*))|
```

Comment:
Can it be made more generic ex- _[a-z ]+_ rather _BRCA1_ to be able to extract mutation and exonic location irrespective of what comes ex- BRCA2 rather BRCA1?

Reply:
See commit [a76e454](https://github.com/NHSDigital/data_management_system/pull/56/commits/a76e45453449614aceec720cd7730fb4d5d7efc9) and [9c4ffb7](https://github.com/NHSDigital/data_management_system/pull/56/commits/9c4ffb747ead4916222b1cb06408c269bc9e2d7a)